common/service/dynamicconfig/config.go

```diff
@@ -39,26 +39,23 @@ func NewCollection(client Client, logger bark.Logger) *Collection {
 // code
 type Collection struct {
 	client Client
-	logger bark.Logger
+	logger log.Logger
 	keys   *sync.Map
 }
 
 func (c *Collection) logNoValue(key Key, err error) {
 	_, loaded := c.keys.LoadOrStore(key, struct{}{})
 	if !loaded {
-		c.logger.Debugf("Failed to fetch key: %s from dynamic config with err: %s", key.String(), err.Error())
+		c.logger.Debug("Failed to fetch key: %s from dynamic config with err: %s", tag.Key(key.String()), tag.Error(err))
```

Comment:
Is key guaranteed to be non nil?

Reply:
Yes I believe so, from the calling path (otherwise we already have panics)